app/models/work_packages/scopes/relatable.rb

```diff
@@ -40,33 +40,114 @@ module Relatable
+      # PARENT relationship, work packages directly related to any of the descendants are added as well. Since work packages
+      # related to the queried work packages only by a path containing both FOLLOWS and PRECEDES relationships are valid targets,
+      # these are filtered out as well. It would be possible to optimize the CTE here to not have them included
+      # in the first place.
+      #
+      # The CTE has the following columns:
+      # * id - the id of the work packages currently related. This is the result of the CTE.
+      # * from_(hierarchy/from_id/to_id) - booleans to prevent that the CTE returns back the path calculated in the previous
+      #                                    iteration.
+      # * origin - boolean to indicate whether the work package is the queried for work package or its ancestor/descendant
+      #            (only descendant for PARENT). Such a work package is never a valid target.
+      #            Additionally, it is used to limit proceeding in both directions for FOLLOWS/PRECEDES.
+      # * includes_(from_relation/to_relation) - booleans about the direction (from_id -> to_id or to_id -> from_id) of the path
+      #                                          (the relations followed).
+      #                                          This is relevant for a queried for PARENT relation. In that case, relations need
+      #                                          to be followed from the queried for work package (and its descendants) in both
+      #                                          directions. But only the direction taken from that origin needs to be followed
+      #                                          henceforth.
+      # * includes_hierarchy - boolean indicating that the last relation taken was a hierarchy relation. For a queried for
+      #                        PARENT relation, whenever that is the case, the work package is a valid relation target although
+      #                        it appears in the CTE.
       def relatable(work_package, relation_type)
         return all if work_package.new_record?
 
         scope = not_having_directed_relation(work_package, relation_type)
-                  .not_having_direct_relation(work_package)
-                  .where.not(id: work_package)
+                  .not_having_direct_relation(work_package, relation_type)
+                  .where.not(id: work_package.id)
+
+        # On a parent relationship, explicitly remove the former parent (which might be the current one as well)
+        # from the list of work packages one can relate to. This is not strictly necessary since it would not
+        # cause faulty relationships but doing it removes the parent from places where it should not show up,
+        # e.g. in an auto completer.
+        if relation_type == Relation::TYPE_PARENT && work_package.parent_id_was
+          scope = scope.where.not(id: work_package.parent_id_was)
+        end
```

Comment:
Why not move this into `not_having_direct_relation`?

Reply:
I did this now though the implementation has changed quite a bit in a7c3914b77a7b43c9a0bf5c19df2c10969889db1. I am not sure if it fits in the newly created `not_having_potential_tree_relation` just on the basis of the name but it makes the code easier to read.